ophyd/device.py

```diff
@@ -207,22 +257,33 @@ def __repr__(self):
         suffix = (repr(self.suffix) if self.suffix
                   else '')
 
-        component_class = self.cls.__name__
+        component_class = self.cls.__name__ if self.cls else ""
```

Comment:
What situation is being handled with this change here? Someone passing in a weird value for cls in `__init__`?

Reply:
I had a branch where I allowed `cls` to be derived from the annotation alone, e.g., `Cpt[EpicsSignal]()` would be valid. I thought that would be deemed too magical so I removed it from the proposal.

This is an unintended change left in from the above; I'll remove it.

Reply:
Resolved in 1acca934a43fc8ca5577a421e9d082cb70d10966